### ISSUE TITLE
Get tailscale approved for APP on android

### ISSUE THREAD

**Issue by danderson** [tailscale/tailscale#1727]
The google advanced protection program is a thing that locks down your google account quite hard, and among other things pops up dire "be really careful installing this app!" on android when you install stuff from the Play Store that hasn't been specifically marked as APP-safe.

https://support.google.com/accounts/answer/9764949?hl=en has instructions to apply for inclusion in the APP program as an approved app (which removes those warnings). I think we can make a reasonable argument for Tailscale being useful and non-harmful to APP users, so we should probably try to apply and see what Google thinks.

**Comment by DentonGentry** [comment]
I submitted the form to start the process on 4/20/2021.

**Comment by DentonGentry** [comment]
I received the automated response from Google Forms shortly after submission. It looks like it submits to a spreadsheet.

**Comment by DentonGentry** [comment]
The trouble with submitting a Google Form is that you can't tell the difference between them having a backlog they're working through versus the form being a blackhole that somebody set up who has since left the company and nobody else knows about.

**Comment by DentonGentry** [comment]
No love from Google APP yet.

**Comment by bradfitz** [comment]
I'm told we're now approved.

Leaving open for somebody to verify.

**Comment by DentonGentry** [comment]
I created a new GMail account and enrolled it in APP. Look in 1password for "Google Advanced Protection Program account"
Advanced Protection requires two hardware FIDO tokens, both of which I have. If we need to use the account for further testing we'll need to add more tokens somehow. It looks like online 2FA like Duo isn't supported, only hardware tokens.

https://accounts.google.com/ shows Advanced Protection is enabled on the account.

<img width="899" alt="Screen Shot 2021-05-22 at 9 42 35 AM" src="https://user-images.githubusercontent.com/1140756/119234316-2c16d080-bae2-11eb-930c-12cbd0f8e528.png">

----

I factory reset a Pixel 1 running Android 9 and signed in with only the APP account. It required the 2FA token to be inserted.

The first time I tried to install Tailscale it failed with a 403 error, however a short time later I got a notification that Google Play Protect had finished scanning the device. After that, the Tailscale app installed and was able to join the Tailnet.

However I've been able to install a number of other apps from the Play Store: clocks, games, and deliberately sought out an app with only 100 total downloads. All of them install and run. I didn't get any warnings nor did I have to go to *See details* and then *Continue to install* as described in https://support.google.com/accounts/answer/9764949?hl=en

The APP docs say, *"You aren't able to use your Google Account with some apps and services that require access to sensitive data like your emails and Google Drive."* So maybe the app verification is needed before accessing other Google services, but doesn't block just any arbitrary app from being installed.

I'm going to close the bug since the primary objective has been verified: Google accounts enrolled in the Advanced Protection Program can install and use the Tailscale app on Android.

![Screenshot_20210522-092007](https://user-images.githubusercontent.com/1140756/119233687-105dfb00-badf-11eb-8510-1cd3ed55a10f.png)
